File: Python/A/567A-linelandmail.py
n = int(input())
cities = list(map(int, input().split()))
print(cities[1] - cities[0], cities[-1] - cities[0])
for i in range(1, n - 1):
    cost1, cost2 = cities[-1] - cities[i], cities[i] - cities[0]
    max_cost = max(cost1, cost2)
    cost1, cost2 = cities[i] - cities[i - 1], cities[i + 1] - cities[i]
    min_cost = min(cost1, cost2)
    print(min_cost, max_cost)
print(cities[-1] - cities[-2], cities[-1] - cities[0])

File: C++/A/567A-linelandmail.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> cities(n);
    for (int i = 0; i < n; i++)
        cin >> cities[i];
    cout << cities[1] - cities[0] << ' ' << cities.back() - cities[0] << '\n';
    for (int i = 1; i < n - 1; i++)
    {
        int cost1 = cities.back() - cities[i], cost2 = cities[i] - cities[0];
        int max_cost = max(cost1, cost2);
        cost1 = cities[i] - cities[i - 1], cost2 = cities[i + 1] - cities[i];
        int min_cost = min(cost1, cost2);
        cout << min_cost << ' ' << max_cost << '\n';
    }
    cout << cities.back() - cities[n - 2] << ' ' << cities.back() - cities[0] << '\n';
}
